### PR TITLE
Fixes #19368 - If a float custom field is empty, display it as an emp…

### PR DESCRIPTION
**Pull request by bgamrat**
…ty string, rather than zero.  Courtesy of Game Creek Video.

### CODE REVIEW

File: core/cfdefs/cfdef_standard.php [hunk 1]
```diff
@@ -218,7 +218,10 @@ function cfdef_print_numeric( $p_value ) {
  * @param string $p_value The custom field value.
  */
 function cfdef_print_float( $p_value ) {
-	echo (float)$p_value;
+	if (is_numeric($p_value)) {
```
Comment (vboctor) [inline core/cfdefs/cfdef_standard.php]: I would change this to:
```
echo is_numeric( $p_value ) ? (float)$p_value : '';
```

**Comment by vboctor** [comment]
Please also move the "Fixed #...." to the body of the comment rather than the title and keep the comment first line / title for what is being fixed.

```
Show empty string for empty float custom fields

Fixes #19368
```

**Comment by atrol** [comment]
> Fixes #19368

#19368 is : `custom field (type: float) default value not work.` and has been fixed in version 1.3.0-beta.2.

Please enter a new issue for it.

File: core/cfdefs/cfdef_standard.php [hunk 2]
```diff
@@ -218,7 +218,10 @@ function cfdef_print_numeric( $p_value ) {
...
+		echo (float)$p_value;
+	}
+	echo '';
```
Comment (atrol) [inline core/cfdefs/cfdef_standard.php]: Do we need to echo an empty string?

File: core/cfdefs/cfdef_standard.php (hunk 1 above)
Comment (bgamrat) [inline core/cfdefs/cfdef_standard.php]: Changed as recommended

**Comment by cproensa** [comment]
As far as i know, the same happens for "numeric" field type

**Comment by vboctor** [comment]
Opened appropriate issues and merged via d24fa5e2c52850ea67935b37d53ec2a453b4596d and 84c9dc08ec43ebde3bbcd63528a5bcc32548e780.

Thanks @bgamrat for your contribution.

**Pull request closed without merging**